packages/dd-trace/src/config.js

```diff
@@ -806,31 +544,285 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
+    }
+    this._setString(opts, 'iast.telemetryVerbosity', options.iastOptions && options.iastOptions.telemetryVerbosity)
+    this._setBoolean(opts, 'isCiVisibility', options.isCiVisibility)
+    this._setBoolean(opts, 'logInjection', options.logInjection)
+    this._setString(opts, 'lookup', options.lookup)
+    this._setBoolean(opts, 'openAiLogsEnabled', options.openAiLogsEnabled)
+    this._setValue(opts, 'peerServiceMapping', options.peerServiceMapping)
+    this._setBoolean(opts, 'plugins', options.plugins)
+    this._setString(opts, 'port', options.port)
+    this._setBoolean(opts, 'profiling.enabled', options.profiling)
+    this._setString(opts, 'protocolVersion', options.protocolVersion)
+    if (options.remoteConfig) {
+      this._setValue(opts, 'remoteConfig.pollInterval', maybeFloat(options.remoteConfig.pollInterval))
+    }
+    this._setBoolean(opts, 'reportHostname', options.reportHostname)
+    this._setBoolean(opts, 'runtimeMetrics', options.runtimeMetrics)
+    this._setUnit(opts, 'sampleRate', coalesce(options.sampleRate, options.ingestion.sampleRate))
+    const ingestion = options.ingestion || {}
+    this._setValue(opts, 'sampler.rateLimit', coalesce(options.rateLimit, ingestion.rateLimit))
+    this._setString(opts, 'service', options.service || tags.service)
+    this._setString(opts, 'site', options.site)
+    if (options.spanAttributeSchema) {
+      this._setString(opts, 'spanAttributeSchema', validateNamingVersion(options.spanAttributeSchema))
+    }
+    this._setBoolean(opts, 'spanRemoveIntegrationFromService', options.spanRemoveIntegrationFromService)
+    this._setBoolean(opts, 'startupLogs', options.startupLogs)
     this._setTags(opts, 'tags', tags)
+    this._setBoolean(opts, 'telemetry.logCollection', options.iastOptions &&
+    (options.iastOptions === true || options.iastOptions.enabled === true))
+    this._setBoolean(opts, 'traceId128BitGenerationEnabled', options.traceId128BitGenerationEnabled)
+    this._setBoolean(opts, 'traceId128BitLoggingEnabled', options.traceId128BitLoggingEnabled)
+    this._setString(opts, 'version', options.version || tags.version)
+  }
+
+  _isCiVisibility () {
+    return coalesce(
+      this.options.isCiVisibility,
+      this._defaults['isCiVisibility']
+    )
+  }
+
+  _isCiVisibilityItrEnabled () {
+    return coalesce(
+      process.env.DD_CIVISIBILITY_ITR_ENABLED,
+      true
+    )
+  }
+
+  _getHostname () {
+    const DD_CIVISIBILITY_AGENTLESS_URL = process.env.DD_CIVISIBILITY_AGENTLESS_URL
+    const url = DD_CIVISIBILITY_AGENTLESS_URL ? new URL(DD_CIVISIBILITY_AGENTLESS_URL)
+      : getAgentUrl(this._getTraceAgentUrl(), this.options)
+    const DD_AGENT_HOST = coalesce(
+      this.options.hostname,
+      process.env.DD_AGENT_HOST,
+      process.env.DD_TRACE_AGENT_HOSTNAME,
+      '127.0.0.1'
+    )
+    return DD_AGENT_HOST || (url && url.hostname)
+  }
+
+  _getSpanComputePeerService () {
```

Comment:
For this, it's fine (and probably simpler/better) to:
* Compute a `peerServiceValue` configuration item in the manner you introduced (i.e. _applyDefaults to undefined, _applyEnv and _applyOptions)
* Compute a `peerServiceSet` configuration item in the manner you introduced (making _applyDefaults false, _applyEnv true if the env var is set, _applyOptions true if the key is in the options)
* Make the final `spanComputePeerService` decision from both of the above + `config.spanAttributeSchema`, which is already computed properly

Reply:
`config.peerServiceValue` and `peerServiceSet` are not set until `_merge` is called, which gets called _after_ `_applyCalculated`

Reply:
In this case, either we need one more step to set derived values after we've computed the merge, or the derivation belongs in plugin code in some form. Either way, the existing code is fine, so for FI we're good as is 👍